### PR TITLE
MAGECLOUD-4100: Introduce DI component

### PR DESCRIPTION
**Pull request by oshmyheliuk**
### Description

Laravel DI container replaced with Symfony.
https://magento2.atlassian.net/browse/MAGECLOUD-4100

### Manual testing scenarios

1. Push Magento on the cloud.
2. Check that the deployment process finishe without errors.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: config/services/build/generate.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8" ?>
```
Comment (oshmyheliuk) [inline config/services/build/generate.xml]: Files generate.xml and transfer.xml just examples of how ece-tools commands can be configured by Symfony DI.
But at the moment this has no sense as they are configured in scneario*.xml files.

File: src/App/Container.php [hunk 2]
```diff
@@ -30,105 +31,54 @@ class Container implements ContainerInterface
...
-        $this->container->singleton(
-            \Magento\MagentoCloud\Config\Database\ConfigInterface::class,
-            \Magento\MagentoCloud\Config\Database\MergedConfig::class
+        $containerBuilder->set(SystemList::class, $systemList);
+        $containerBuilder->setDefinition(SystemList::class, new Definition(SystemList::class));
+
+        $containerBuilder->set(Composer::class, $this->createComposerInstance($systemList));
+        $containerBuilder->setDefinition(Composer::class, new Definition(Composer::class));
+
+        $definition = new Definition();
```
Comment (shiftedreality) [inline src/App/Container.php]: This is defined in `services.xml`

**Review by shiftedreality — CHANGES_REQUESTED** [review]

File: config/services.xml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        <service id="Magento\MagentoCloud\Step\Deploy\InstallUpdate\Update" autowire="false" />
+        <service id="Magento\MagentoCloud\Step\Deploy\PreDeploy" autowire="false" />
+        <service id="Magento\MagentoCloud\Step\StepComposite" autowire="false" />
+        <service id="Magento\MagentoCloud\Step\StepException" autowire="false" />
+        <service id="Magento\MagentoCloud\Step\ValidateConfiguration" autowire="false" />
+        <service id="Magento\MagentoCloud\Step\Build\BackupData" autowire="false" />
+        <service id="Magento\MagentoCloud\Service\ServiceMismatchException" autowire="false" />
+        <service id="Magento\MagentoCloud\Shell\Process" autowire="false" />
+        <service id="Magento\MagentoCloud\Shell\ProcessException" autowire="false" />
+
```
Comment (shiftedreality) [inline config/services.xml]: Excessive empty line

File: src/App/Container.php [hunk 4]
```diff
@@ -30,105 +31,54 @@ class Container implements ContainerInterface
...
+        $this->container = $containerBuilder;
+    }
+
+    /**
+     * @param SystemList $systemList
+     * @return Composer
+     */
+    private function createComposerInstance(SystemList $systemList): Composer
+    {
+        $composerFactory = new \Composer\Factory();
```
Comment (shiftedreality) [inline src/App/Container.php]: Please use import from namespace

Comment (oshmyheliuk) [reply src/App/Container.php]: done

File: config/services.xml (hunk 3 above)
Comment (oshmyheliuk) [reply config/services.xml]: removed

File: src/App/Container.php (hunk 2 above)
Comment (oshmyheliuk) [reply src/App/Container.php]: Agree, removed

**Comment by shiftedreality** [comment]
Please fix Travis CI static/unit tests

**Review by shiftedreality — APPROVED** [review]

**Review by BaDos — APPROVED** [review]

**Comment by oshmyheliuk** [comment]
QA done in automated tests.

**Pull request merged**